worklist/views_helper_functions/store_articles.py

```diff
@@ -16,26 +16,25 @@ def store_added_articles(worklist_object, articles):
 
     # Mapping article titles into IDs
     for article in articles:
-        titles.append(str(article['name']))
+        if article['name'] != '':
+            titles.append(article['name'])
```

Comment:
I'd prefer to do the validation of article titles only once (we're doing it twice in this function).

Also, I think it would make sense to do it in the view that calls this function so by the time we get here we know the data is (supposed to be) well-formed. It's probably simpler to validate upfront when we receive the data than to pass around data that may-or-may-not be validated in our controller/model layers.

Sometimes we won't be able to do that, but for simple cases like "article names shouldn't be empty" I think it does make sense to do it in the view.

Reply:
I've removed the double validation. But I feel that the validation part should be in the function itself making it reusable across multiple modules. Otherwise every module that'll be using it needs to do those checks.